test/local/sign_routes.js

```diff
@@ -43,40 +43,96 @@ test(
         }
       },
       query: {}
-    }
-    var signRoutes = makeRoutes({
-      config: {
-        memcache: {
-          address: '127.0.0.1:11211',
-          idle: 100
-        }
-      },
-      log: mockLog
     })
 
-    return new P(function (resolve) {
-      getRoute(signRoutes, '/certificate/sign')
-        .handler(mockRequest, resolve)
-    })
-    .then(
-      function () {
-        t.equal(mockLog.activityEvent.callCount, 1)
-        t.equal(mockLog.activityEvent.args[0].length, 3)
-        t.equal(mockLog.activityEvent.args[0][0], 'account.signed')
-        t.equal(mockLog.activityEvent.args[0][1], mockRequest)
-        t.deepEqual(mockLog.activityEvent.args[0][2], {
+    test('without service', function (t) {
+      return runTest({
+        db: mockDB,
+        log: mockLog
+      }, mockRequest, function () {
+        t.equal(mockDB.createDevice.callCount, 1, 'db.createDevice was called once')
+        var args = mockDB.createDevice.args[0]
+        t.equal(args.length, 3, 'db.createDevice was passed one argument')
+        t.equal(args[0], mockRequest.auth.credentials.uid, 'first argument was uid')
+        t.equal(args[1], mockRequest.auth.credentials.tokenId, 'second argument was sessionTokenId')
+        t.deepEqual(args[2], {}, 'third argument was empty object')
+
+        t.equal(mockLog.activityEvent.callCount, 1, 'log.activityEvent was called once')
+        args = mockLog.activityEvent.args[0]
+        t.equal(args.length, 3, 'log.activityEvent was passed three arguments')
+        t.equal(args[0], 'account.signed', 'first argument was event name')
+        t.equal(args[1], mockRequest, 'second argument was request obect')
+        t.deepEqual(args[2], {
           uid: mockRequest.auth.credentials.uid.toString('hex'),
           account_created_at: mockRequest.auth.credentials.accountCreatedAt,
-          device_id: mockRequest.auth.credentials.deviceId.toString('hex')
-        })
-      },
-      function () {
-        t.fail('request should have succeeded')
-      }
-    )
+          device_id: null
```

Comment:
Fair point. I was asserting the existing behaviour from [line 107](https://github.com/mozilla/fxa-auth-server/pull/1299/files#diff-de8109276e632a0c1520d779a2558e5fR107) of `lib/routes/sign` without considering the bigger picture (which is, generally I think, a weakness of mine). Will fix.